File: 121/solution.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_val = max_val = 10000
        res = 0
        for price in prices:
            if price < min_val:
                min_val = max_val = price
            elif price > max_val:
                max_val = price
                res = max(res, price - min_val)
        return res

File: 121/Solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int minVal = 10000;
        int maxVal = 10000;
        int res = 0;
        for (int price: prices) {
            if (price < minVal) {
                maxVal = price;
                minVal = price;
            } else if (price > maxVal) {
                maxVal = price;
                res = Math.max(res, price - minVal);
            }
        }
        return res;
    }
}